src/System.Management.Automation/engine/Subsystem/DscSubsystem/ICrossPlatformDsc.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.ObjectModel;
+using System.Collections.Generic;
+using System.Management.Automation.Internal;
+using System.Management.Automation.Language;
+
+namespace System.Management.Automation.Subsystem
+{
+    /// <summary>
+    /// Interface for implementing a cross platform desired state configuration component.
+    /// </summary>
+    public interface ICrossPlatformDsc : ISubsystem
```

Comment:
I see the `Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscCache` type is used in `SemanticChecks.cs` and `Tokenizer.cs` solely, without the `ICrossPlatformDsc` subsystem in a `if/else` check. Is that an oversight? How are we going to replace the old DSC code completely without covering those uses?

Reply:
Updated. Those places are using essentially string constants that are unlikely to be different throughout different DSC versions.
But thinking more about this, it is probably better to move those to the subsystem interface as well - I've added `IsSystemResourceName` and `IsDefaultModuleNameForMetaConfigResource` to subsystem interface and used those with `is subsystem registered` check in `SemanticChecks.cs` and `Tokenizer.cs`.